amd_openvx_extensions/amd_custom/source/custom_layer.cpp

```diff
@@ -0,0 +1,244 @@
+      ERROR_CHECK_STATUS(vxCopyScalar((vx_scalar)parameters[2], &data->backend, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+
+    // Create custom lib and run setup
+    data->custom_handle = CustomCreate((CustomFunctionType)customFuntion);
+    ERROR_CHECK_CUSTOM_STATUS(CustomSetup(data->custom_handle, data->input_desc, data->output_desc, 
+                              (customBackend)data->backend, (customStream)data->hipstream));
+
+    ERROR_CHECK_STATUS(vxSetNodeAttribute(node, VX_NODE_LOCAL_DATA_PTR, &data, sizeof(data)));
+    return VX_SUCCESS;
+}
+
+static vx_status VX_CALLBACK uninitializeCustomLayer(vx_node node, const vx_reference *parameters, vx_uint32 num)
+{
+    CustomLayerLocalData * data = NULL;
+    ERROR_CHECK_STATUS(vxQueryNode(node, VX_NODE_LOCAL_DATA_PTR, &data, sizeof(data)));
+    ERROR_CHECK_CUSTOM_STATUS(CustomShutdown(data->custom_handle));
+    if (data->pCustomParameterArray)
+        delete data->pCustomParameterArray;
+    delete data;
+
+    return VX_SUCCESS;
+}
+
+vx_status publishCustomLayer(vx_context context)
+{
+    // add kernel to the context with callbacks
+    vx_kernel kernel = vxAddUserKernel(context, "com.amd.custom_extension.custom_layer", VX_KERNEL_CUSTOM_LAYER, processCustomLayer, 5, validateCustomLayer, initializeCustomLayer, uninitializeCustomLayer);
+    ERROR_CHECK_OBJECT(kernel);
+
+    // enable GPU buffer access since the kernel_f callback uses GPU buffers instead of host accessible buffers
+    AgoTargetAffinityInfo affinity;
+    vxQueryContext(context, VX_CONTEXT_ATTRIBUTE_AMD_AFFINITY, &affinity, sizeof(affinity));
+#if ENABLE_OPENCL || ENABLE_HIP
+    // enable OpenCL buffer access since the kernel_f callback uses OpenCL buffers instead of host accessible buffers
+    vx_bool enableBufferAccess = vx_true_e;
+    if (affinity.device_type == AGO_TARGET_AFFINITY_GPU)
+        ERROR_CHECK_STATUS(vxSetKernelAttribute(kernel, VX_KERNEL_ATTRIBUTE_AMD_GPU_BUFFER_ACCESS_ENABLE, &enableBufferAccess, sizeof(enableBufferAccess)));
+#endif
+
+    // set kernel parameters
+    ERROR_CHECK_STATUS(vxAddParameterToKernel(kernel, 0, VX_INPUT, VX_TYPE_TENSOR, VX_PARAMETER_STATE_REQUIRED));
+    ERROR_CHECK_STATUS(vxAddParameterToKernel(kernel, 1, VX_INPUT, VX_TYPE_SCALAR, VX_PARAMETER_STATE_REQUIRED));
+    ERROR_CHECK_STATUS(vxAddParameterToKernel(kernel, 2, VX_INPUT, VX_TYPE_SCALAR, VX_PARAMETER_STATE_OPTIONAL));
+    ERROR_CHECK_STATUS(vxAddParameterToKernel(kernel, 3, VX_INPUT, VX_TYPE_ARRAY, VX_PARAMETER_STATE_OPTIONAL));
+    ERROR_CHECK_STATUS(vxAddParameterToKernel(kernel, 4, VX_OUTPUT, VX_TYPE_TENSOR, VX_PARAMETER_STATE_REQUIRED));
+
+    // finalize and release kernel object
+    ERROR_CHECK_STATUS(vxFinalizeKernel(kernel));
+    ERROR_CHECK_STATUS(vxReleaseKernel(&kernel));
+
+    return VX_SUCCESS;
+}
+
+VX_API_ENTRY vx_node VX_API_CALL vxCustomLayer(vx_graph graph, vx_tensor inputs, vx_uint32 function, vx_uint32 custom_backend, vx_array custom_parameters, vx_tensor outputs)
+{
```

Comment:
formatting: move the { to the previous line

Reply:
done